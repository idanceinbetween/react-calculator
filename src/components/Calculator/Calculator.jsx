import React, {Component} from 'react';
import './Calculator.css';
import Display from '../Display/Display';
import Keypad from '../Keypad/Keypad';

class Calculator extends Component {

    state = {
        displayValue: '0',
        numbers: ['9','8','7','6','5','4','3','2','1','0','.','ce'],
        operators: ['/', 'x', '-', '+'],
        selectedOperator: '',
        storedValue: ''
    }

    callOperator = () => console.log('Call Operator')
    setOperator = () => console.log('Set Operator')
    updateDisplay = () => console.log('Update Display')

    render() {
        const { displayValue, numbers, operators} = this.state

        return (
            <div className="calculator-container">
                <Display displayValue={displayValue}/>
                <Keypad
                    callOperator={this.callOperator}
                    numbers={numbers}
                    operators={operators}
                    setOperator={this.setOperator}
                    updateDisplay={this.updateDisplay}
                />
            </div>
        );
    }
}

export default Calculator;