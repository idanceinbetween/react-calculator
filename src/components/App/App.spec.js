import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import Calculator from "../Calculator/Calculator";

describe('App', ()=>{
    let wrapper;
    beforeEach(()=> wrapper = shallow(<App />));

    it('renders correctly', () => expect(wrapper).toMatchSnapshot());

    it('this test will pass', ()=>{
        expect(true).toBeTruthy;
    })

    it('renders a div', ()=>{
        expect(wrapper.find('div').length).toEqual(1);
    })

    it('has a Calculator component', () => {
        expect(wrapper.containsMatchingElement(<Calculator />)).toEqual(true);
    })
})