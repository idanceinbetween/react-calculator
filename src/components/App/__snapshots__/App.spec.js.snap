// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly with Snapshot 1`] = `ShallowWrapper {}`;
